bankcraft/model.py

```diff
@@ -11,19 +13,25 @@ def __init__(self, num_people = 10, INITIAL_MONEY=1000,
         
         self.num_people = num_people
         self.schedule = RandomActivation(self)
-    
+
+        # adding a network
+        self.G = nx.erdos_renyi_graph(self.num_people, 0.5)
+        self.grid = NetworkGrid(self.G)
         
         # Adding PeopleAgents
-        for i in range(self.num_people):
+        for i,node in enumerate(self.G.nodes()):
             person = Person(i, self, INITIAL_MONEY, 
                  SPENDING_PROB, SPENDING_AMOUNT, SALARY)
             self.schedule.add(person)
 
+            # add agent to a random node
+            self.grid.place_agent(person, node)
+
         self.datacollector = DataCollector(
              # collect agent money
-            agent_reporters = {"Money": lambda a: a.money}
-    
-        )
+            agent_reporters = {"Money": lambda a: a.money},
+            # collect model state
+            model_reporters = {'graph': lambda m: m.grid.get_all_cell_contents(), })
 
```

Comment:
I think it is better to place everyone on a grid, which represents the geographic space the agents inhabit, and have the network as a property of the agents. Keeping an adjacency list on each person, for example. It would be nice to have functionality added to roll up the adjacency lists into a full network for visualization/analysis though.

Reply:
I changed this part to have a social network graph as a property of the model.